src/hapi/pipelines/database/food_price.py

```diff
@@ -0,0 +1,116 @@
+"""Populate the WFP market table."""
+
+from logging import getLogger
+from typing import Dict, List
+
+from dateutil.relativedelta import relativedelta
+from hapi_schema.db_food_price import DBFoodPrice
+from hdx.scraper.utilities.reader import Read
+from hdx.utilities.dateparse import parse_date
+from sqlalchemy.orm import Session
+
+from ..utilities.logging_helpers import add_missing_value_message
+from .base_uploader import BaseUploader
+from .currency import Currency
+from .metadata import Metadata
+from .wfp_commodity import WFPCommodity
+from .wfp_market import WFPMarket
+
+logger = getLogger(__name__)
+
+
+class FoodPrice(BaseUploader):
+    def __init__(
+        self,
+        session: Session,
+        datasetinfo: Dict[str, str],
+        countryiso3s: List[str],
+        metadata: Metadata,
+        currency: Currency,
+        commodity: WFPCommodity,
+        market: WFPMarket,
+    ):
+        super().__init__(session)
+        self._datasetinfo = datasetinfo
+        self._countryiso3s = countryiso3s
+        self._metadata = metadata
+        self._currency = currency
+        self._commodity = commodity
+        self._market = market
+
+    def populate(self):
+        logger.info("Populating WFP price table")
+        reader = Read.get_reader("hdx")
+        headers, iterator = reader.read(datasetinfo=self._datasetinfo)
+        datasetinfos = []
+        for country in iterator:
+            countryiso3 = country["countryiso3"]
+            if countryiso3 not in self._countryiso3s:
+                continue
+            dataset_name = country["url"].split("/")[-1]
+            datasetinfos.append(
+                {
+                    "dataset": dataset_name,
+                    "format": "csv",
+                    "admin_single": countryiso3,
+                }
+            )
+        warnings = set()
+        errors = set()
+        for datasetinfo in datasetinfos:
+            headers, rows = reader.read(datasetinfo)
+            hapi_dataset_metadata = datasetinfo["hapi_dataset_metadata"]
+            hapi_resource_metadata = datasetinfo["hapi_resource_metadata"]
+            self._metadata.add_hapi_metadata(
+                hapi_dataset_metadata, hapi_resource_metadata
+            )
+            countryiso3 = datasetinfo["admin_single"]
+            dataset_name = hapi_dataset_metadata["hdx_stub"]
+            resource_id = hapi_resource_metadata["hdx_id"]
+            for row in rows:
+                market = row["market"]
+                if "#" in market:
+                    continue
```

Comment:
why?

Reply:
The first row has HXL hashtags

Reply:
But now you mention it I should just do a next()